std/protocols/permutation.asm

```diff
@@ -3,17 +3,76 @@ use std::array::fold;
 use std::array::len;
 use std::convert::int;
 use std::check::assert;
-use std::field::modulus;
+use std::check::panic;
+use std::field::known_field;
+use std::field::KnownField;
+use std::math::fp2::Fp2Value;
+use std::math::fp2::Fp2Expr;
+use std::math::fp2::add_ext;
+use std::math::fp2::sub_ext;
+use std::math::fp2::mul_ext;
+use std::math::fp2::unpack_ext;
+use std::math::fp2::next_ext;
+use std::math::fp2::eval_ext;
+use std::math::fp2::expr_ext;
+use std::math::fp2::inv_ext;
 
 let is_first: col = |i| if i == 0 { 1 } else { 0 };
 
 // Get two phase-2 challenges to use in all permutation arguments.
-// Note that this assumes that globally no other challenge of these IDs is used.
-let alpha: expr = challenge(0, 1);
-let beta: expr = challenge(0, 2);
+// Note that this assumes that globally no other challenge of these IDs is used,
+// and that challenges for multiple permutation arguments are re-used.
```

Comment:
I think that's fine, but not sure what the best practice is here?

Reply:
We'll fix that once we have the global identities transformer.